### PR TITLE
Create uhome.js

### PR DESCRIPTION
**Pull request by AlexBolshov**
Added KDE TWV as new device

### CODE REVIEW

**Comment by Koenkk** [comment]
Can you provide a link to this device?

**Comment by AlexBolshov** [comment]
https://uhome.systems/Smart-valve.html

File: devices/kdesv.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+const exposes = require('../lib/exposes');
+const reporting = require('../lib/reporting');
+const ota = require('../lib/ota');
+const e = exposes.presets;
+
+module.exports = [
+    {
+        zigbeeModel: ['TWV'],
+        model: 'TWV',
+        vendor: 'KDE',
```
Comment (Koenkk) [inline devices/kdesv.js:12]: The brand seems to be UHome and not KDE?

Comment (AlexBolshov) [reply devices/kdesv.js]: It was pervious name on the site. Yes, you are right, sorry, that was my mistake.

Comment (AlexBolshov) [reply devices/kdesv.js]: I changed vendor field and renamed "file.js" for convenience.

**Comment by AlexBolshov** [comment]
Keep or delete commits where I fixed my mistakes? Please let me know what else can I do.

File: devices/uhome.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+const ota = require('../lib/ota');
+const e = exposes.presets;
+
+module.exports = [
+    {
+        zigbeeModel: ['TWV'],
+        model: 'TWV',
+        vendor: 'UHome',
+        description: 'UHome Smart Valve',
+        ota: ota.zigbeeOTA,
```
Comment (Koenkk) [inline devices/uhome.js:14]: Do we have an OTA available for this device?

**Comment by Koenkk** [comment]
> Keep or delete commits where I fixed my mistakes? Please let me know what else can I do.

No need to, I will squash the commits.

Can you also provide the data/database.db entry of this device?

File: devices/uhome.js (hunk 2 above)
Comment (AlexBolshov) [reply devices/uhome.js:14]: Yes OTA cluster working. 
I've missed something with OTA?

Comment (AlexBolshov) [reply devices/uhome.js:14]: > Can you also provide the data/database.db entry of this device?

Sure, where can I find example of this entry?

**Comment by AlexBolshov** [comment]
Did you mean this? https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html#_4-add-device-picture-to-zigbee2mqtt-io-documentation After resolving this pull request, I thought it would be the next step.

**Comment by Koenkk** [comment]
You can find the database.db in the zigbee2mqtt `data` directory.

File: devices/uhome.js (hunk 2 above)
Comment (Koenkk) [reply devices/uhome.js:14]: But we also need to have the OTA file itself this need to be added to https://github.com/koenkk/zigbee-ota

Comment (AlexBolshov) [reply devices/uhome.js:14]: Of course I can do that. But this is a separate repository, I wanted to do it after this was done.
Sorry for my poor git repo knowledge, do I need to pull request the OTA repository with a separate pull request or can I do it in this pull request? Same with data/database.db.
Also there is no database.db in Koenkk/zigbee2mqtt/data, only configuration.yaml.

Comment (Koenkk) [reply devices/uhome.js:14]: Can you make the pull request for the OTA first?

Comment (AlexBolshov) [reply devices/uhome.js:14]: Sorry for the late reply, there was an blackout. I'll try to do it this weekend if I have the light.

Comment (AlexBolshov) [reply devices/uhome.js:14]: The valve currently does not have a "Manufacturer ID". I am currently using the test ID "123".
Is it possible to upload such an OTA image (with manufacturer ID "123") to the repository?
If not, can we pause the OTA image upload for now and keep the OTA features?
Or just remove it from configuration file?

**Comment by AlexBolshov** [comment]
{"id":2,"type":"EndDevice","ieeeAddr":"0xf4ce361f4f64e764","nwkAddr":38868,"manufId":4660,"manufName":"UHome","powerSource":"Battery","modelId":"TWV","epList":[1],"endpoints":{"1":{"profId":260,"epId":1,"devId":1,"inClusterList":[0,6,1,3],"outClusterList":[25],"clusters":{"genBasic":{"attributes":{"swBuildId":""}},"genPowerCfg":{"attributes":{"batteryPercentageRemaining":40}},"genOnOff":{"attributes":{"onOff":1}}},"binds":[{"cluster":1,"type":"endpoint","deviceIeeeAddress":"0x00124b00256c4c91","endpointID":1},{"cluster":6,"type":"endpoint","deviceIeeeAddress":"0x00124b00256c4c91","endpointID":1}],"configuredReportings":[{"cluster":1,"attrId":33,"minRepIntval":3600,"maxRepIntval":62000,"repChange":0},{"cluster":6,"attrId":0,"minRepIntval":0,"maxRepIntval":3600,"repChange":0}],"meta":{}}},"appVersion":1,"stackVersion":10,"hwVersion":11,"dateCode":"20221209","swBuildId":"","zclVersion":8,"interviewCompleted":true,"meta":{"configured":-2071105341},"lastSeen":1668969485340,"defaultSendRequestWhen":"immediate"}

from data/database.db 

Did you mean this?

File: devices/uhome.js (hunk 2 above)
Comment (Koenkk) [reply devices/uhome.js:14]: Can you send me the OTA file?

Comment (AlexBolshov) [reply devices/uhome.js:14]: I made pull request with TWV image in OTA repository so you can see how it looks.

**Comment by Koenkk** [comment]
Perfect, thanks!

**Comment by AlexBolshov** [comment]
Thank you!!!

**Pull request merged**